met-web/src/components/engagement/view/widgets/EventsWidget.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
+import { MetPaper, MetHeader2, MetBody } from 'components/common';
+import { Grid, Link, Skeleton, Divider } from '@mui/material';
+import { Widget } from 'models/widget';
+import { Event, EVENT_TYPE } from 'models/event';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { When } from 'react-if';
+import { getEvents } from 'services/widgetService/EventService';
+import { formatDate } from 'components/common/dateHelper';
+
+interface EventsWidgetProps {
+    widget: Widget;
+}
+const EventsWidget = ({ widget }: EventsWidgetProps) => {
+    const dispatch = useAppDispatch();
+    const [isLoading, setIsLoading] = useState(true);
+    const [events, setEvents] = useState<Event[]>([]);
+
+    const fetchEvents = async () => {
+        try {
+            const events = await getEvents(widget.id);
+            setEvents(events);
+            setIsLoading(false);
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while fetching Engagement widgets information',
+                }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        fetchEvents();
+    }, [widget]);
+
+    if (isLoading) {
+        return (
+            <MetPaper elevation={1} sx={{ padding: '1em' }}>
+                <Grid container justifyContent="flex-start" spacing={3}>
+                    <Grid item xs={12}>
+                        <MetHeader2>
+                            <Skeleton variant="rectangular" />
+                        </MetHeader2>
+                    </Grid>
+                    <Grid item xs={12}>
+                        <Skeleton variant="rectangular" height="10em" />
+                    </Grid>
+                    <Grid item xs={12}>
+                        <Skeleton variant="rectangular" height="10em" />
+                    </Grid>
+                </Grid>
+            </MetPaper>
+        );
+    }
+
+    if (events.length === 0) {
+        return null;
+    }
+
+    return (
+        <MetPaper elevation={1} sx={{ padding: '1em', minHeight: '12em' }}>
```

Comment:
I would prefer breaking virtual and in person to separate components. and do a Case switch like in DocumentSwitch.tsx. This way they would be completely separate for maintenance. It's not a blocker though. I'll add approval and leave it to you.

Reply:
Ok I think that is a better idea for the future of this component. So I will work on that and get it merged. Thank you.